src/test/unittest/ut_signal.c

```diff
@@ -40,12 +40,31 @@
 /*
  * On Windows, Access Violation exception does not raise SIGSEGV signal.
  * The trick is to catch the exception and... call the signal handler.
- *
- * XXX - add support for registering more than one signal/exception handler
  */
 
-static void (*Sa_handler) (int signum);
-static int Signum;
+struct signal_handler {
```

Comment:
I believe the lock should not be necessary, adding signal handlers from multiple threads is not a good thing to do. Adding the lock should not affect anything, but we can add it. Preferably with a trylock and an assert, so that we don't ever try to do this while multithreading.

As for **sa_mask**, AFAIK it is defined per registered handler, which is registered per process, so it shouldn't be held per thread.